query-graphs/src/tree-rendering.ts

```diff
@@ -581,11 +582,52 @@ export function drawQueryTree(target: HTMLElement, treeData: TreeDescription) {
         zoomBehavior.translateTo(baseSvg, x, y);
     }
 
+    // Scale the graph so that it can fit nicely on the screen
+    function fitGraphScale() {
+        // Get the bounding box of the main SVG element, we are interested in the width and height
+        const bounds = baseSvg.node().getBBox();
+        // Get the size of the container of the main SVG element
+        const parent = baseSvg.node().parentElement;
+        const fullWidth = parent.clientWidth;
+        const fullHeight = parent.clientHeight;
+        // Let's find the scale factor
+        // Thinking of only the X dimension, we need to make the width match the fullWidth, Equation:
+        //   width * xfactor = fullWidth
+        // Solving for `xfactor` we have: xfactor = fullWidth/width
+        const xfactor: number = fullWidth / bounds.width;
+        // Similarly, we would find out that: yfactor = fullHeight/height;
+        const yfactor: number = fullHeight / bounds.height;
+        // Most likely, the X and Y factor are different. We use the minimum of them to avoid cropping
+        let scaleFactor: number = Math.min(xfactor, yfactor);
+        // Add some padding so that the graph is not touching the edges
+        const paddingPercent = 0.9;
+        scaleFactor = scaleFactor * paddingPercent;
+        console.log("Scale factor", scaleFactor);
```

Comment:
Do we still need the debug output?
Here and other places

Reply:
removing...